rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -27,6 +28,33 @@ namespace rosbag2_cpp
 {
 namespace readers
 {
+namespace details
+{
+std::vector<std::string> resolve_relative_paths(
+  const std::string & base_folder, std::vector<std::string> relative_files, const int version = 4)
+{
+  auto base_path = rcpputils::fs::path(base_folder);
+  if (version < 4) {
+    // In older rosbags (version <=3) relative files are prefixed with the rosbag folder name
+    base_path = rcpputils::fs::path(base_folder).parent_path();
```

Comment:
If we try to play the bag from the same directory level as the bag directory, then `base_path` is empty and the assertion at L42 fails.

I think this is a bug in rcpputils though, since a path to the current directory should "exist", hopefully.

Reply:
Proposed fix in https://github.com/ros2/rcpputils/pull/73